database/mysql.go

```diff
@@ -148,37 +178,61 @@ func CreateTestMySQLDB(t *testing.T) *TestMySQLDB {
 		t.Skip("Docker not enabled")
 	}
 
-	config, container, err := RunMySQLDockerInstance(&DatabaseConfig{})
-	if err != nil {
-		t.Fatal(err)
+	err := SharedMySQL.Start()
+	require.NoError(t, err)
+
+	dbName, err := CreateTemporaryDatabase(SharedMySQL.Config)
+	require.NoError(t, err)
+
+	dbConfig := &DatabaseConfig{
+		DatabaseName: dbName,
+		MySQL:        SharedMySQL.Config.MySQL,
 	}
 
 	logger := log.NewNopLogger()
-
 	ctx, cancelFunc := context.WithCancel(context.Background())
-
-	db, err := NewAndMigrate(ctx, logger, *config)
+	db, err := NewAndMigrate(ctx, logger, *dbConfig)
 	if err != nil {
-		container.Close()
 		t.Fatal(err)
 	}
 
 	// Don't allow idle connections so we can verify all are closed at the end of testing
 	db.SetMaxIdleConns(0)
 
 	return &TestMySQLDB{
-		DB:        db,
-		container: container,
-		shutdown:  cancelFunc,
-		t:         t,
+		DB:       db,
+		name:     dbName,
+		shutdown: cancelFunc,
+		t:        t,
 	}
 }
 
-func RunMySQLDockerInstance(config *DatabaseConfig) (*DatabaseConfig, *dockertest.Resource, error) {
-	if config.DatabaseName == "" {
-		config.DatabaseName = "test"
+// We connect as root to MySQL server and create database with random name to
+// run our migrations on it later.
+func CreateTemporaryDatabase(config *DatabaseConfig) (string, error) {
+	dsn := fmt.Sprintf("%s:%s@%s/", "root", config.MySQL.Password, config.MySQL.Address)
+	db, err := sql.Open("mysql", dsn)
+	if err != nil {
+		return "", err
+	}
+	defer db.Close()
+
+	dbName := "test" + base.ID()
```

Comment:
Instead of `base.ID()` which is 20 random characters; using `t.Name()` may be preferrable?

Reply:
Right now we create unique database per test, next week we will use one database per test binary + DB transactions. Also, I don't want this function to depend on `t`.